compiler-plugin-tests/src/test/java/io/ballerina/stdlib/http/compiler/staticcodeanalyzer/StaticCodeAnalyzerTest.java

```diff
@@ -18,99 +18,164 @@
+                    "Avoid permissive Cross-Origin Resource Sharing",
+                    RuleKind.VULNERABILITY);
+            Assertions.assertRule(
+                    rules,
+                    "ballerina/http:3",
+                    "Server-side requests should not be vulnerable to traversing attacks",
+                    RuleKind.VULNERABILITY);
+
+            // validate the issues
+            List<Issue> issues = testRunner.getIssues();
+            int index = 0;
+
+            if (rule == AVOID_DEFAULT_RESOURCE_ACCESSOR) {
+                Assert.assertEquals(issues.size(), 8);
+                Assertions.assertIssue(issues, index++, "ballerina/http:1", "service.bal",
+                        19, 19, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:1", "service.bal",
+                        23, 23, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:1", "service_class.bal",
+                        21, 21, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:1", "service_class.bal",
+                        25, 25, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:1", "service_object.bal",
+                        20, 20, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:1", "service_object.bal",
+                        21, 21, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:1", "service_object.bal",
+                        26, 26, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index, "ballerina/http:1", "service_object.bal",
+                        27, 27, Source.BUILT_IN);
+            } else if (rule == AVOID_PERMISSIVE_CORS) {
+                Assert.assertEquals(issues.size(), 7);
+                Assertions.assertIssue(issues, index++, "ballerina/http:2", "service.bal",
+                        20, 20, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:2", "service.bal",
+                        27, 27, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:2", "service_class.bal",
+                        23, 23, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:2", "service_object.bal",
+                        20, 20, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:2", "service_object.bal",
+                        28, 28, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index++, "ballerina/http:2", "service_object.bal",
+                        36, 36, Source.BUILT_IN);
+                Assertions.assertIssue(issues, index, "ballerina/http:2", "service_object.bal",
+                        44, 44, Source.BUILT_IN);
+            } else if (rule == AVOID_TRAVERSING_ATTACKS) {
```

Comment:
Checking the each element using an `if-else` block inside the loop is not a good practice. The method is too long as well. Can we break it to separate methods, maybe?